File: domain/src/main/java/com/example/app/movie/domain/Movie.java
package com.example.app.movie.domain;

import com.example.app.movie.type.MovieGenre;
import com.example.app.movie.type.MovieRating;
import com.example.app.movie.type.MovieStatus;
import lombok.Builder;

import java.time.LocalDate;
import java.util.List;

@Builder
public record Movie(
        Long id,
        String title,
        String description,
        MovieStatus status,
        MovieRating rating,
        MovieGenre genre,
        String thumbnail,
        int runningTime,
        LocalDate releaseDate,
        List<Showtime> showtimes,
        List<Theater> theaters
){

}

File: core/src/main/kotlin/kr/spartacodingclub/cinema/core/domain/Movie.kt
package kr.spartacodingclub.cinema.core.domain

import java.time.LocalDate

data class Movie(
    val id: Long,
    val title: String,
    val rating: MovieRating,
    val releaseDate: LocalDate,
    val thumbnailUrl: String,
    val runningTime: Int,
    val genre: Genre
)
